test/client/api/directoryAdmin.unittest.ts

```diff
@@ -0,0 +1,326 @@
+import {Identity} from 'openfin/_v2/main';
+
+import {updateAppDirectory, AppDirectory, APP_DIRECTORY_STORAGE_TAG, StoredDirectoryShardMap, UpdateAppDirectoryMigrationHandler} from '../../../src/client/api/directoryAdmin';
+import {createMockFin, createMockApplication, getterMock} from '../../mocks';
+import {createFakeIdentity, createFakeApp, createFakeUrl} from '../../demo/utils/fakes';
+import {Application} from '../../../src/client/main';
+
+type TestParam = [string, UpdateAppDirectoryMigrationHandler, StoredDirectoryShardMap];
+
+const fin = createMockFin();
+
+const fakeApp1 = createFakeApp();
```

Comment:
would it look less verbose if you kept these in an array?

Reply:
Probably not what you had in mind, but I've reduced the duplication here. Resistant to just keeping these in arrays for readability of the tests themselves